fawltydeps/main.py

```diff
@@ -211,53 +212,77 @@ def build_parser() -> argparse.ArgumentParser:
     return parser
 
 
-def main() -> int:
-    """Command-line entry point."""
-    parser = build_parser()
-    args = parser.parse_args()
-    settings = Settings.config(config_file=args.config_file).create(args)
-
-    logging.basicConfig(level=logging.WARNING - 10 * settings.verbosity)
-
-    if args.generate_toml_config:
-        print_toml_config(settings, sys.stdout)
-        return 0
-
-    try:
-        analysis = Analysis.create(settings)
-    except UnparseablePathException as exc:
-        return parser.error(exc.msg)  # exit code 2
+def assign_exit_code(analysis: Analysis) -> int:
+    """
+    Assign exit code base in the analysis results.
 
-    # Exit codes:
-    # 0 - success, no problems found
-    # 1 - an exception propagates (this should not happen)
-    # 2 - command-line parsing error (see above)
-    # 3 - undeclared dependencies found
-    # 4 - unused dependencies found
+    Exit codes:
+    0 - success, no problems found
+    1 - an exception propagates (this should not happen)
+    2 - command-line parsing error (see above)
+    3 - undeclared dependencies found
+    4 - unused dependencies found
+    """
     exit_code = 0
     if analysis.is_enabled(Action.REPORT_UNDECLARED) and analysis.undeclared_deps:
         exit_code = 3
     elif analysis.is_enabled(Action.REPORT_UNUSED) and analysis.unused_deps:
         exit_code = 4
 
+    return exit_code
```

Comment:
Nit:

I'd slightly prefer if <A>: return 3, if <B>: return 4, then just return 0 at the end (i.e., `elif` becomes `if`, and rather than initalizing and reassigning `exit_code`, there are just the raw values), but OK either way